06-building-decays.md

```diff
@@ -0,0 +1,142 @@
+If you access the code for `StdAllNoPIDsPions`, you can see that in it the algorithm for creating the particles is instantiated and the map of the `DataOnDemand` service is updated via the `updateDoD` function.
+
+Once we have the input pions and kaons, we can combine them to build a D0 by means of the `CombineParticles` algorithm.
+This algorithm performs the combinatorics for us according to a given decay descriptor and puts the resulting particle in the TES, allowing also to apply some cuts on them:
+
+ - `DaughtersCuts` is a dictionary that maps each child particle we want to cut on with a selection LoKi particle functor (or combination of functors) which results in a `bool`. Optionally, one can specify also a `Preambulo` property that allows us to make imports, preprocess functors, etc (more on this in the [LoKi functors](https://lhcb.github.io/first-analysis-steps/06-loki-functors.html) lesson). For example:
+
+```python
+d0_daughters = {'pi-': '(PT > 750*MeV) & (P > 4000*MeV) & (MIPCHI2DV(PRIMARY) > 4)',
+                'K+': '(PT > 750*MeV) & (P > 4000*MeV) & (MIPCHI2DV(PRIMARY) > 4)'}
+```
+
+ - `CombinationCut` is a particle array LoKi functor (note the `A` prefix, see more [here](https://twiki.cern.ch/twiki/bin/view/LHCb/LoKiHybridFilters#Particle_Array_Functors)) that is given as input the array of particles in a single combination, in our case a kaon and a pion. This cut is performed the vertex fit, so typically can be used to save CPU cycles by performing some sanity cuts such as `AMAXDOCA` or `ADAMASS` (do you know what they do?) before the vertex fit is performed:
+ 
+```python
+d0_comb = "(AMAXDOCA('') < 0.2*mm) & (ADAMASS('D0') < 100*MeV)"
+```
+ 
+ - `MotherCut` is a selection LoKi particle functor that acts on the parent particle after the vertex fit, which allows to further filter on those variables that require the particle to be built:
+
+```python
+d0_mother = "(VFASPF(VCHI2/VDOF)< 9) & (BPVDIRA > 0.9997) & (ADMASS('D0') < 70*MeV)"
+```
+
+So we can build a combiner as
+
+```python
+from Configurables import CombineParticles
+d0 = CombineParticles("Combine_D0",
+                      Decay='([D0 -> pi- K+]CC)',
+                      DaughtersCuts=d0_daughters,
+                      CombinationCut=d0_comb,
+                      MotherCut=d0_mother)
+```
+
+Now we have to build a `Selection` out of it so we can later on put all pieces together:
+
+```python
+from PhysSelPython.Wrappers import Selection
+d0_sel = Selection("Sel_D0",
+                   Algorithm=d0,
+                   RequiredSelections=[Pions, Kaons])
+```
+
+We may now apply some selection on the soft pion before combining it with the D0.
+Instead of using `DaughtersCuts`, we can make use of the `FilterDesktop` algorithm, which takes a TES location and filters the particles inside according to a given LoKi functor in the `Code` property:
```

Comment:
OK! This I can do, but I thought it would be repetitive. There's a reason
to use filter desktop (reusability), it's just that in this example it's
not possible to show it.